apps/mobile/src/components/ThreadCard/ThreadCard.js

```diff
@@ -38,42 +45,60 @@ export default function ThreadCard ({ message, currentUser, onPress, participant
   )
 }
 
-export function lastMessageCreator (message, currentUser, participants) {
+export function lastMessageCreator (message, currentUser, participants = []) {
   const { t } = useTranslation()
-  if (get('id', message.creator) === currentUser.id) return `${t('You')}: `
+  
+  if (!message?.creator?.id || !currentUser?.id) return ''
+  
+  if (message.creator.id === currentUser.id) return `${t('You')}: `
   if (participants.length <= 2) return ''
-  return (find(p => p.id === get('id', message.creator), participants)?.name || t('Deleted User')) + ': '
+  
+  const creator = find(p => p?.id === message.creator.id, participants)
+  return (creator?.name || t('Deleted User')) + ': '
 }
 
-export function threadNames (names) {
+export function threadNames (names = []) {
   const { t } = useTranslation()
-  const l = names.length
+  
+  const validNames = names.filter(name => name && typeof name === 'string')
+  const l = validNames.length
+  
   switch (l) {
     case 0:
       return t('You')
     case 1:
     case 2:
-      return names.join(', ')
+      return validNames.join(', ')
     default:
-      return `${names.slice(0, 1).join(', ')} ${t('and')} ${l - 1} ${t('other')}${l > 2 ? 's' : ''}`
+      return `${validNames.slice(0, 1).join(', ')} ${t('and')} ${l - 1} ${t('other')}${l > 2 ? 's' : ''}`
   }
 }
 
-export function ThreadAvatars ({ avatarUrls }) {
-  const count = avatarUrls.length
+export function ThreadAvatars ({ avatarUrls = [] }) {
+  const validAvatarUrls = avatarUrls.filter(Boolean)
+  const count = validAvatarUrls.length
+  
+  if (count === 0) {
+    return (
+      <View style={styles.threadAvatars}>
+        <Avatar avatarUrl={null} style={styles.firstThreadAvatar} />
```

Comment:
avatarUrl={null}? why would we even have this?

Reply:
So Avatar component knows to use the default avatar. This is for the edge-case of when there are no valid avatar urls